File: mobile/strip.py
import os
import re

def remove_comments(text):
    # Regex to match strings (group 1) and comments (group 2)
    # Multi-line strings, single-line strings, and raw variations
    string_pattern = r'r?"""(?:\\.|[^\\])*?"""|r?\'\'\'(?:\\.|[^\\])*?\'\'\'|r?"(?:\\.|[^"\\])*"|r?\'(?:\\.|[^\'\\])*\''
    comment_pattern = r'(/\*[\s\S]*?\*/|//.*)'
    pattern = re.compile(f'({string_pattern})|{comment_pattern}')
    
    def replacer(match):
        if match.group(2) is not None:
            return ''
        return match.group(1)
        
    result = pattern.sub(replacer, text)
    
    # remove purely empty lines that only had comments
    lines = [line for line in result.split('\n') if line.strip() != '']
    return '\n'.join(lines) + '\n'

for root, dirs, files in os.walk('lib'):
    for file in files:
        if file.endswith('.dart'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            new_content = remove_comments(content)
            with open(path, 'w', encoding='utf-8') as f:
                f.write(new_content)

File: mobile/strip.js
const fs = require('fs');
const path = require('path');

function removeComments(text) {
    const stringPattern = /r?"""(?:\\.|[^\\])*?"""|r?'''(?:\\.|[^\\])*?'''|r?"(?:\\.|[^"\\])*"|r?'(?:\\.|[^'\\])*'/g;
    const commentPattern = /\/\*[\s\S]*?\*\/|\/\/.*/g;
    
    // Replace comments with empty string, keep strings
    let result = text.replace(new RegExp(`${stringPattern.source}|${commentPattern.source}`, 'g'), (match) => {
        if (match.startsWith('//') || match.startsWith('/*')) {
            return '';
        }
        return match;
    });
    
    return result.split('\n').filter(line => line.trim() !== '').join('\n') + '\n';
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.dart')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let newContent = removeComments(content);
            fs.writeFileSync(fullPath, newContent, 'utf8');
        }
    }
}

walkDir('lib');
